### ISSUE TITLE
Deprecate this.el and this.els

### ISSUE THREAD

**Issue by patrick-steele-idem** [marko-js/marko#809]
In the next minor version of Marko, we will have full support for multiple root elements (that includes dynamic root elements, text nodes at the root, etc.). This makes`this.el` and `this.els` problematic since `this.el` implies there is only one root HTML element and `this.els` is bad since it implies the top-level nodes for a UI component are only HTML elements. Instead, of using `this.el` and `this.els`, a developer should instead assign keys and use `this.getEl(key)` or `this.getEls(key)`.

**Comment by riktimmondal** [comment]
Please guide me since i am a beginner i will like to contribute

**Comment by patrick-steele-idem** [comment]
@riktimmondal Thank you for your interest in contributing! Another developer, @DylanPiercey, has started on a task of providing the ability to add debug messages and deprecation warnings in the development build of marko. We want deprecation warnings and debug messages to be stripped out as part of the production build. Once Dylan completes that task then its only a matter of adding a deprecation warning inside the getter method for `this.el` and `this.els`.  We can find you another task if you are interested in contributing (I'll go through and /cc you on a few that might be of interest) :)

**Comment by mlrawlings** [comment]
Blocked by #871

**Comment by DylanPiercey** [comment]
Fixed in https://github.com/marko-js/marko/pull/916
